src/jit/morph.cpp

```diff
@@ -3684,107 +3618,69 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
 #endif
 #endif
 
-#ifndef LEGACY_BACKEND
+            fgArgTabEntryPtr newArgEntry;
             if (lateArgsComputed)
             {
-                if (argEntry->isNonStandard)
-                {
-                    // This is a non-standard register argument - possibly update it in the table
-                    call->fgArgInfo->RemorphRegArg(argIndex, argx, args, argEntry->regNum, size, argAlign);
-                    argIndex++;
-                    continue;
-                }
+                // This is a register argument - possibly update it in the table
+                newArgEntry = call->fgArgInfo->RemorphRegArg(argIndex, argx, args, nextRegNum, size, argAlign);
```

Comment:
You can probably remove the assignment and just change this to 
call->fgArgInfo->RemorphRegArg(argIndex, argx, args, nextRegNum, size, argAlign); 
and remove the declaration of newArgEEntry above.